recon/src/protocol.rs

```diff
@@ -102,8 +103,8 @@ where
 /// Message that the initiator produces
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum InitiatorMessage<K: Key, H: AssociativeHash> {
-    /// Declares interestes to the responder.
-    InterestRequest(Vec<RangeOpen<K>>),
+    /// Declares interests to the responder.
+    InterestRequest((u64, Vec<RangeOpen<K>>)),
```

Comment:
Should we wrap any of these values in structs? Would make adding fields/behavior easier possibly...  but new messages with v2 or whatever might be better